wagtail/wagtailadmin/tests/test_rich_text.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import absolute_import, unicode_literals
+
+import unittest
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.test import TestCase
+from django.test.utils import override_settings
+
+from wagtail.tests.utils import WagtailTestUtils
+from wagtail.wagtailcore.models import Page
+
+from wagtail.tests.testapp.rich_text import CustomRichTextArea
+from wagtail.wagtailadmin.rich_text import get_rich_text_editor_widget, HalloRichTextArea
+
+
+class TestGetRichTextEditorWidget(TestCase):
+    @override_settings()
+    def test_default(self):
+        # Simulate the absence of a setting
+        if hasattr(settings, 'WAGTAILADMIN_RICH_TEXT_EDITORS'):
+            del settings.WAGTAILADMIN_RICH_TEXT_EDITORS
+
+        self.assertIsInstance(get_rich_text_editor_widget(), HalloRichTextArea)
+
+    @override_settings(WAGTAILADMIN_RICH_TEXT_EDITORS={
+        'default': {
+            'WIDGET': 'wagtail.tests.testapp.rich_text.CustomRichTextArea'
+        },
+    })
+    def test_overridden_default_editor(self):
+        self.assertIsInstance(get_rich_text_editor_widget(), CustomRichTextArea)
+
+    @override_settings(WAGTAILADMIN_RICH_TEXT_EDITORS={
+        'custom': {
+            'WIDGET': 'wagtail.tests.testapp.rich_text.CustomRichTextArea'
+        },
+    })
+    def test_custom_editor_without_default(self):
+        self.assertIsInstance(get_rich_text_editor_widget('custom'), CustomRichTextArea)
+
+    @override_settings(WAGTAILADMIN_RICH_TEXT_EDITORS={
+        'default': {
+            'WIDGET': 'wagtail.wagtailadmin.rich_text.HalloRichTextArea'
+        },
+        'custom': {
+            'WIDGET': 'wagtail.tests.testapp.rich_text.CustomRichTextArea'
+        },
+    })
+    def test_custom_editor_with_default(self):
+        self.assertIsInstance(get_rich_text_editor_widget(), HalloRichTextArea)
+        self.assertIsInstance(get_rich_text_editor_widget('custom'), CustomRichTextArea)
+
+
+@override_settings()
+class TestDefaultRichText(TestCase, WagtailTestUtils):
+
+    def setUp(self):
+        # Find root page
+        self.root_page = Page.objects.get(id=2)
+
+        self.login()
+
+        # Simulate the absence of a setting
+        if hasattr(settings, 'WAGTAILADMIN_RICH_TEXT_EDITORS'):
+            del settings.WAGTAILADMIN_RICH_TEXT_EDITORS
+
+    def tearDown(self):
+        from wagtail.tests.testapp.models import DefaultRichBlockFieldPage
+        from wagtail.tests.testapp.models import DefaultRichTextFieldPage
```

Comment:
I have spent lot of time to figure out why some tests fail when I run test case, but I still don't know how to fix it (how to reset some property caches).

Looks like part of problem in [this cache](https://github.com/torchbox/wagtail/blob/v1.4.4/wagtail/wagtailadmin/edit_handlers.py#L290) and in [this property](https://github.com/torchbox/wagtail/pull/2416/files#diff-de5efc8fde48bf2b10e49731bab27710R327)... Probably somewhere else.

@kaedroho, do you have any idea how to test editor with different settings?